### ISSUE TITLE
Latest NixOS update (unstable) breaks Cosmic deployment?

### ISSUE THREAD

**Issue by ams-123** [lilyinstarlight/nixos-cosmic#419]
Since a large NixOS update some 24 hours back, I'm unable to upgrade either of the systems I have running Cosmic. Both give the same error:

```
nixos-rebuild switch --flake . --use-remote-sudo
building the system configuration...
error:
       … while calling the 'head' builtin

         at /nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source/lib/attrsets.nix:1575:11:

         1574|         || pred here (elemAt values 1) (head values) then
         1575|           head values
             |           ^
         1576|         else

       … while evaluating the attribute 'value'

         at /nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source/lib/modules.nix:816:9:

          815|     in warnDeprecation opt //
          816|       { value = addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          817|         inherit (res.defsFinal') highestPrio;

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'targetPlatforms' missing

       at /nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source/pkgs/build-support/rust/build-rust-package/default.nix:163:7:

          162|       meta.platforms or lib.platforms.all
          163|       rustc.targetPlatforms;
             |       ^
          164|   };
error: Recipe `deploy` failed on line 10 with exit code 1
```

**Comment by lilyinstarlight** [comment]
Fix is already building in #418. Note that this wouldn't have been an issue for anyone using the current flake input pattern recommended in the readme: <https://github.com/lilyinstarlight/nixos-cosmic/blob/a1e3c25b1d6fb0a88a5bbb61cc4bd502439a68ff/readme.md?plain=1#L18>

**Comment by ams-123** [comment]
Thanks @lilyinstarlight . Missed that flake change somehow. It's working now.
